### PR TITLE
[Deprecate] Send a body with the patch request

### LINKED ISSUE #52: pod trunk deprecate net error
### Report

🌈

* What did you do?
 * `pod trunk deprecate <podname>`
* What did you expect to happen?
 * pod to deprecate
* What happened instead?
 * Network related error, looks like passing a hash rather than a string.

### Stack

```
   CocoaPods : 1.0.0.beta.2
        Ruby : ruby 2.2.3p173 (2015-08-18 revision 51636) [x86_64-darwin15]
    RubyGems : 2.4.5.1
        Host : Mac OS X 10.11.2 (15C50)
       Xcode : 7.2 (7C68)
         Git : git version 2.7.0
Ruby lib dir : /Users/ms/.rbenv/versions/2.2.3/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 0be149713bb7f6b100e0ab0d66d362a2a2a43efa
```

### Plugins

```
cocoapods-deintegrate : 1.0.0.beta.1
cocoapods-plugins     : 1.0.0.beta.1
cocoapods-search      : 1.0.0.beta.1
cocoapods-stats       : 1.0.0.beta.3
cocoapods-trunk       : 1.0.0.beta.1
cocoapods-try         : 1.0.0.beta.2
```

### Error

```
NoMethodError - undefined method `bytesize' for #<Hash:0x007fc934049908>
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/2.2.0/net/http/generic_request.rb:182:in `send_request_with_body'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/2.2.0/net/http/generic_request.rb:120:in `exec'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/2.2.0/net/http.rb:1412:in `block in transport_request'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/2.2.0/net/http.rb:1411:in `catch'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/2.2.0/net/http.rb:1411:in `transport_request'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/2.2.0/net/http.rb:1384:in `request'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/nap-1.0.0/lib/rest/request.rb:186:in `block in perform'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/2.2.0/net/http.rb:853:in `start'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/nap-1.0.0/lib/rest/request.rb:186:in `perform'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/nap-1.0.0/lib/rest/request.rb:195:in `perform'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/nap-1.0.0/lib/rest.rb:68:in `patch'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cocoapods-trunk-1.0.0.beta.1/lib/pod/command/trunk.rb:45:in `create_request'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cocoapods-trunk-1.0.0.beta.1/lib/pod/command/trunk.rb:28:in `request_url'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cocoapods-trunk-1.0.0.beta.1/lib/pod/command/trunk.rb:36:in `request_path'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cocoapods-trunk-1.0.0.beta.1/lib/pod/command/trunk/deprecate.rb:33:in `deprecate'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cocoapods-trunk-1.0.0.beta.1/lib/pod/command/trunk/deprecate.rb:28:in `run'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/claide-1.0.0.beta.1/lib/claide/command.rb:312:in `run'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cocoapods-1.0.0.beta.2/lib/cocoapods/command.rb:48:in `run'
/Users/ms/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cocoapods-1.0.0.beta.2/bin/pod:44:in `<top (required)>'
/Users/ms/.rbenv/versions/2.2.3/bin/pod:23:in `load'
/Users/ms/.rbenv/versions/2.2.3/bin/pod:23:in `<main>'
```

### PR DESCRIPTION
**Pull request by segiddins**
Closes #52.

\c @orta

### CODE REVIEW

**Comment by orta** [comment]
Yep! This looks like I'd expect

**Comment by orta** [comment]
:+1:

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+## Master
+
+##### Bug Fixes
+
+* Send a body with the `PATCH` request to deprecate a pod.
```
Comment (kylef) [inline CHANGELOG.md]: Kinda too late to this one, but as a user this means nothing to an average user. The changelog shouldn't talk about implementation details if it doesn't need them but instead explain what user flow or user action is now fixed.

**Comment by segiddins** [comment]
@kylef make a pull request improving it then

**Pull request merged**